### PR TITLE
Integrated publickey import and removal in iosxr_user

### PR DESCRIPTION
**Pull request by sebasdoes**
##### SUMMARY

Also see issue https://github.com/ansible/ansible/issues/31965, after discussion on IRC we decided to integrate this in the iosxr_user module and create a pull request from there.

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

iosxr_user

##### ANSIBLE VERSION

```
(venv) root@DESKTOP-BM1CR3D:~/Documents/T-mobile/git/netops-ansible/staging/playbooks# ansible --version
ansible 2.5.0 (devel 40e7de32d5) last updated 2017/10/25 09:19:10 (GMT +200)
  config file = /mnt/c/Users/Hoofdgebruiker/Documents/T-mobile/git/netops-ansible/staging/playbooks/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /mnt/c/Users/Hoofdgebruiker/Documents/T-mobile/git/ansible/lib/ansible
  executable location = /mnt/c/Users/Hoofdgebruiker/Documents/T-mobile/git/ansible/bin/ansible
  python version = 3.5.2 (default, Sep 14 2017, 22:51:06) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION

* Tests still need to be made.
* We need to work around the dependency in my code on module.params['host'] and module.params['username']. I did not know how to do this.

```

```

### CODE REVIEW

**Comment by sebasdoes** [comment]
I need to fix my issue with module.params['host'] and username before I can pass these tests.

File: lib/ansible/modules/network/iosxr/iosxr_user.py [hunk 1]
```diff
@@ -275,6 +357,18 @@ def main():
                            mutually_exclusive=mutually_exclusive,
                            supports_check_mode=True)
 
+    if module.params['publickeyfile']:
+        if not HAS_B64:
+            module.fail_json(
+                msg='library base64 is required but does not appear to be '
```
Comment (gundalow) [inline lib/ansible/modules/network/iosxr/iosxr_user.py]: Needs adding to the DOCUMENTATION block
```
requirements:
  - base64 when using I(publickeyfile)
  - paramiko when using I(publickeyfile)
```

http://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-block

Comment (sebasdoes) [reply lib/ansible/modules/network/iosxr/iosxr_user.py]: Done, although in the newer version it's going to be called public_key_contents.
I'll push again once I've fixed some errors. Added some integration tests as well.

**Comment by sebasdoes** [comment]
Okay, it's finally compliant with the checks. 
Please let me know if I need to change more!

**Comment by sebasdoes** [comment]
I've also added the ability to add multiple groups.

**Pull request merged**